snuba/clickhouse/native.py

```diff
@@ -6,6 +6,8 @@
 from typing import Iterable, Mapping, Optional
 from uuid import UUID
 
+import sentry_sdk
```

Comment:
I would not start spans in this file.
In snuba we cannot instrument all code paths, because when we instrument a code path it will generate events, which will go to sentry and come back to snuba. Everything on the query path is fine, this does not create a loop. Everything on the ingestion path is not fine, otherwise we would create a new event to ingest for each event we ingest. 
Everything else is kind of undetermined, we never decided upon, this includes replacers, migration script, cleanup scripts and so on.

By creating a span, right now we are not creating a transaction, so unless we create a transaction, no event is created and the span is basically thrown away.
https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/hub.py#L444-L452

Though I would not rely on this behavior (not to send transactions) for code paths where we may not want to generate instrumentation. Just to be on the safe side.
This file may be used in such paths (specifically replacers). So I would remove the span creation from here since we achieve a quite similar result by instrumenting web/query
at least till when we set up instrumentation properly, by creating transactions, in the code paths other than querying.

Reply:
The sampling would suppress a firestorm of bad ju-ju. But, okay. There's not much in here that adds time.